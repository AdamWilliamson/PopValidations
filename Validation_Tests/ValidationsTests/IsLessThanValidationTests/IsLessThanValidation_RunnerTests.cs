using ApprovalTests;
using FluentAssertions;
using System;
using System.Threading.Tasks;
using PopValidations;
using PopValidations_Tests.TestHelpers;
using PopValidations_Tests.ValidationsTests.GenericTestableObjects;
using Xunit;

namespace PopValidations_Tests.ValidationsTests.IsLessThanValidationTests;

public class IsLessThan_NoError_TestingValidator : AbstractValidator<NonNullAllFieldTypesDto>
{
    public IsLessThan_NoError_TestingValidator()
    {
        Describe(x => x.Integer).IsLessThan(int.MaxValue);
        Describe(x => x.String).IsLessThan(new string(char.MaxValue, 100));
        Describe(x => x.Decimal).IsLessThan(decimal.MaxValue);
        Describe(x => x.Double).IsLessThan(double.MaxValue);
        Describe(x => x.Short).IsLessThan(short.MaxValue);
        Describe(x => x.Long).IsLessThan(long.MaxValue);
    }
}

public class IsLessThan_AllErrored_TestingValidator : AbstractValidator<NonNullAllFieldTypesDto>
{
    public IsLessThan_AllErrored_TestingValidator()
    {
        Describe(x => x.Integer).IsLessThan(int.MaxValue, options => options.WithErrorMessage("This Integer is not less then MaxValue"));
        Describe(x => x.String).IsLessThan(new string(char.MaxValue, 100));
        Describe(x => x.Decimal).IsLessThan(decimal.MaxValue);
        Describe(x => x.Double).IsLessThan(double.MaxValue);
        Describe(x => x.Short).IsLessThan(short.MaxValue);
        Describe(x => x.Long).IsLessThan(long.MaxValue);
    }
}

public class IsLessThanValidation_RunnerTests
{
    [Fact]
    public async Task GivenAValidator_WithNoErrors_ThenNoErrorsAreProduced()
    {
        // Arrange
        var runner = ValidationRunnerHelper.BasicRunnerSetup(new IsLessThan_NoError_TestingValidator());

        // Act
        var validationResult = await runner.Validate(new NonNullAllFieldTypesDto()
        {
            Integer = int.MaxValue/2,
            String = new string('g', 100),
            Decimal = decimal.MaxValue / 2,
            Double = double.MaxValue / 2,
            Short = short.MaxValue/2,
            Long = long.MaxValue / 2
        });

        // Assert
        validationResult.Errors.Should().BeEmpty();
    }

    [Fact]
    public async Task GivenAValidator_WithErrors_ThenEveryFieldHasErrors()
    {
        // Arrange
        var runner = ValidationRunnerHelper.BasicRunnerSetup(new IsLessThan_AllErrored_TestingValidator());

        // Act
        var validationResult = await runner.Validate(new NonNullAllFieldTypesDto());
        var json = JsonConverter.ToJson(validationResult);

        // Assert
        validationResult.Errors.Should().HaveCount(6);
        Approvals.VerifyJson(json);
    }

    [Fact]
    public void GivenAValidator_WhenDescribing_ThenEveryFieldHasDescriptions()
    {
        // Arrange
        var runner = ValidationRunnerHelper.BasicRunnerSetup(new IsLessThan_NoError_TestingValidator());

        // Act
        var description = runner.Describe();
        var json = JsonConverter.ToJson(description);

        // Assert
        description.Results.Should().HaveCount(6);
        Approvals.VerifyJson(json);
    }
}
