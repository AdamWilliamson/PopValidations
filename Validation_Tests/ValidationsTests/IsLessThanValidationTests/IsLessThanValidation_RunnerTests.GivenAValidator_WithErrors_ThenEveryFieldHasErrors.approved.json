{
    "errors": {
        "integer": [
            "Is not less than '2147483647'."
        ],
        "string": [
            "Is not less than '￿￿￿￿￿￿￿￿￿￿￿￿￿￿￿￿￿￿￿￿￿￿￿￿￿￿￿￿￿￿￿￿￿￿￿￿￿￿￿￿￿￿￿￿￿￿￿￿￿￿￿￿￿￿￿￿￿￿￿￿￿￿￿￿￿￿￿￿￿￿￿￿￿￿￿￿￿￿￿￿￿￿￿￿￿￿￿￿￿￿￿￿￿￿￿￿￿￿￿￿'."
        ],
        "decimal": [
            "Is not less than '79228162514264337593543950335'."
        ],
        "double": [
            "Is not less than '1.7976931348623157E+308'."
        ],
        "short": [
            "Is not less than '32767'."
        ],
        "long": [
            "Is not less than '9223372036854775807'."
        ]
    }
}